android/engine/src/main/java/org/smartregister/fhircore/engine/ui/bottomsheet/RegisterBottomSheetView.kt

```diff
@@ -0,0 +1,149 @@
+import org.smartregister.fhircore.engine.ui.theme.DividerColor
+import org.smartregister.fhircore.engine.ui.theme.StatusTextColor
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+@Composable
+fun RegisterBottomSheetView(
+  modifier: Modifier = Modifier,
+  navigationMenuConfigs: List<NavigationMenuConfig>?,
+  registerCountMap: Map<String, Long> = emptyMap(),
+  menuClickListener: (NavigationMenuConfig) -> Unit,
+  onDismiss: () -> Unit
+) {
+  Surface(shape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)) {
+    Column(modifier = modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp)) {
+      Text(
+        text = stringResource(R.string.other_patients),
+        textAlign = TextAlign.Start,
+        fontWeight = FontWeight.Bold,
+        fontSize = 16.sp,
+        modifier = modifier.padding(horizontal = 12.dp, vertical = 16.dp).align(Alignment.Start)
+      )
+      Divider(color = DividerColor, thickness = 1.dp)
+      LazyColumn(
+        contentPadding = PaddingValues(vertical = 8.dp),
+        modifier = Modifier.fillMaxWidth()
+      ) {
+        itemsIndexed(navigationMenuConfigs!!) { index, item ->
+          RegisterListItem(
+            navigationMenuConfig = item,
+            registerCountMap = registerCountMap,
+            menuClickListener = menuClickListener,
+            onDismiss = onDismiss
+          )
+          if (index < navigationMenuConfigs.lastIndex)
+            Divider(color = DividerColor, thickness = 1.dp)
+        }
+      }
+    }
+  }
+}
+
+@Composable
+fun RegisterListItem(
+  modifier: Modifier = Modifier,
+  navigationMenuConfig: NavigationMenuConfig,
+  registerCountMap: Map<String, Long> = emptyMap(),
+  menuClickListener: (NavigationMenuConfig) -> Unit,
+  onDismiss: () -> Unit
+) {
+
+  val action = remember {
+    navigationMenuConfig.actions?.find { it.trigger == ActionTrigger.ON_COUNT }
+  }
+  Row(
+    modifier =
+      modifier
+        .fillMaxWidth()
+        .clickable {
+          // Act on click then dismiss dialog
+          menuClickListener(navigationMenuConfig)
+          onDismiss()
+        }
+        .padding(14.dp)
+  ) {
+    Box(modifier = modifier.wrapContentWidth()) {}
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+      Text(text = navigationMenuConfig.display, modifier = modifier.padding(horizontal = 12.dp))
+      Text(
+        text = registerCountMap[action?.id ?: navigationMenuConfig.id]?.toString() ?: "",
+        textAlign = TextAlign.Start,
+        color = StatusTextColor,
+        fontSize = 13.sp,
+      )
+    }
+  }
+}
+
+@Preview(showBackground = true)
+@ExcludeFromJacocoGeneratedReport
+@Composable
+fun RegisterListItemPreview() {
+  RegisterListItem(
+    navigationMenuConfig =
+      NavigationMenuConfig(id = "TestFragmentTag", display = "All Clients", showCount = true),
```

Comment:
is this temporary, while this PR is worked on? and lines 143-144

Reply:
These are Composable view previews that we use to render the compose view on the IDE during development. Also useful to showcase how the composable view is supposed to be used.